src/content/reference/react/use.md

```diff
@@ -0,0 +1,769 @@
+import "./styles.css";
+
+// TODO: update this example to use
+// the Codesandbox Server Component
+// demo enviornemnt once it is created
+import App from "./App";
+
+const root = createRoot(document.getElementById("root"));
+root.render(
+  <StrictMode>
+    <App />
+  </StrictMode>
+);
+```
+
+```json package.json hidden
+{
+  "dependencies": {
+    "react": "18.3.0-canary-9377e1010-20230712",
+    "react-dom": "18.3.0-canary-9377e1010-20230712",
+    "react-scripts": "^5.0.0",
+    "pretty-bytes": "6.1.0"
+  },
+  "main": "/index.js"
+}
+```
+</Sandpack>
+
+<Solution />
+
+#### Read promises from data fetching libraries {/*read-promises-from-data-fetching-libraries*/}
+
+You can read the values of promises returned from data fetching libraries with the `use` Hook. In this example the function `fetchMessage` is imported from a mock data fetching library. The promise returned by `fetchMessage` is stored in the `messagePromise` state variable and then passed to `use` to read. The result is then displayed to the user. While waiting for the promise to resolve the fallback of the parent Suspense component is displayed.
+
+<Sandpack>
+
+```js App.js active
+"use client";
+
+import { use, useState, Suspense } from "react";
+import { fetchMessage } from "./data-fetching-lib.js";
+
+function MessageContainer() {
+  const [messagePromise] = useState(fetchMessage());
+  return (
+    <Suspense fallback={<p>⌛Downloading message...</p>}>
+      <Message messagePromise={messagePromise} />
+    </Suspense>
+  );
+}
+
+function Message({ messagePromise }) {
+  const messageContent = use(messagePromise);
+  return <p>Here is the message: {messageContent}</p>;
+}
+
+export default function App() {
+  const [show, setShow] = useState(false);
+  function download() {
+    setShow(true);
+  }
+
+  if (show) {
+    return <MessageContainer />;
+  } else {
+    return <button onClick={download}>Download message</button>;
+  }
+}
+```
+
+```js index.js hidden
+// TODO: update to import from stable
+// react instead of canary once the `use`
+// Hook is in a stable release of React
+import React, { StrictMode } from "react";
+import { createRoot } from "react-dom/client";
+import "./styles.css";
+
+// TODO: update this example to use
+// the Codesandbox Server Component
+// demo enviornemnt once it is created
+import App from "./App";
+
+const root = createRoot(document.getElementById("root"));
+root.render(
+  <StrictMode>
+    <App />
+  </StrictMode>
+);
+```
+
+```js data-fetching-lib.js hidden
+export function fetchMessage() {
+  return new Promise((resolve) => setTimeout(resolve, 3000, "⚛️"));
+}
+```
+
+```json package.json hidden
+{
+  "dependencies": {
+    "react": "18.3.0-canary-9377e1010-20230712",
+    "react-dom": "18.3.0-canary-9377e1010-20230712",
+    "react-scripts": "^5.0.0"
+  },
+  "main": "/index.js"
+}
+```
+</Sandpack>
+
+<Solution />
+
+</Recipes>
+
+
+
+### Reading context with `use` {/*reading-context-with-use*/}
```

Comment:
this should probably answer the question somewhere: is there a reason to use useContext over use?

also, examples don't show conditional use

Reply:
"is there a reason to use useContext over use?" probably not except that it's not available in stable. Added a sentence to indicate `use` is preferred over `useContext`

Altered example to demonstrate conditional use